src/AppBundle/Service/TaskManager.php

```diff
@@ -64,10 +63,10 @@ public function assign(Task $task, UserInterface $user, $position = null)
         });
 
         foreach ($tasks as $taskToAssign) {
-            $taskToAssign->assignTo($user, $position++);
-
             $isAssignedToSameUser = $taskToAssign->isAssigned() && $taskToAssign->isAssignedTo($user);
+
             if (!$isAssignedToSameUser) {
+                $taskToAssign->assignTo($user, $position++);
```

Comment:
`TaskManager::assign` is also called when reordering tasks already assigned to the same user. 
`Task::assignTo` needs to be outside the `if`